src/iminuit/cost.py

```diff
@@ -676,6 +789,77 @@ def __init__(self, n, xe, model, verbose):
         super().__init__(describe(model)[1:], self._prepare_data(n), verbose)
 
 
+class BarlowBeestonLite(MaskedCost):
+    """
+    Binned cost function for a template fit with uncertainties on the template.
+
+    Compared to the Beeston-Barlow method, the lite method uses one nuisance
+    parameter per bin instead of one nuisance parameter per component per bin,
+    and replaces the Poisson constraint with a Gaussian on the multiplicative
+    nuisance parameter.
+
+    J.S. Conway, PHYSTAT 2011, https://doi.org/10.48550/arXiv.1103.0354
+    P. Mandrik, https://arxiv.org/abs/1708.07708
+    """
+
+    __slots__ = ("_templates", "_sum_templates")
+
+    def __init__(
+        self, n, templates, name: _tp.Collection[str] = None, verbose: int = 0
+    ):
+        """
+        Initialize cost function with data and model.
+
+        Parameters
+        ----------
+        n : array-like
+            Histogram counts.
+        templates : array-like
+            TODO
+        name : collection of str, optional
+            Parameter names.
+        verbose : int, optional
+            Verbosity level. 0: is no output (default).
+            1: print current args and negative log-likelihood value.
+        """
+        n = np.atleast_1d(n)
+        templates = np.atleast_2d(templates)
+        M = len(templates)
+        if M < 1:
+            raise ValueError("at least one template is required")
+        if n.shape != templates.shape[1:]:
+            raise ValueError("shapes of n and templates do not match")
+        self._templates = templates
+        self._sum_templates = np.sum(templates, axis=1)
+        if name is None:
+            name = [f"x{i}" for i in range(M)]
+        else:
+            if len(name) != M:
+                raise ValueError("number of names must match number of templates")
+        super().__init__(name, n, verbose)
+
+    @Cost.ndata.getter  # type:ignore
+    def ndata(self):
+        """See Cost.ndata."""
+        shape = self._masked.shape
+        return np.prod(shape)
+
+    def _call(self, args):
+        mu = 0
+        mu_var = 0
+        for a, t, nt in zip(args, self._templates, self._sum_templates):
+            f = a / nt
+            mu += f * t
+            var_t = t  # different if weighted templates are used, TODO
+            mu_var += var_t * f**2
+
+        ma = self.mask
+        if ma is not None:
+            mu = mu[ma]
+            mu_var = mu_var[ma]
+        return barlow_beeston_lite_chi2(self._masked, mu, mu_var)
```

Comment:
@AhmedAbdelmotteleb To apply BBL, I need to compute the mu and its variance. `mu` is the sum over the component fraction times the template value in a bin, so `mu_var`, the variance of `mu`, needs to be computed from the Poisson uncertainties of the template weighted by the component fraction. Can you please check that this calculation agrees with how you do it?

Reply:
Yes, that is correct. I will check the calculation now.

Reply:
@HDembinski Your implementation looks correct to me. I will add some suggestions though, hope that's okay :)

Reply:
Perfect, thanks.